compiler/lib/driver.ml

```diff
@@ -579,8 +593,11 @@ let full ~standalone ~wrap_with_fun ~profile ~linkall ~source_map formatter d p
        | O2 -> o2
        | O3 -> o3)
     +> exact_calls profile
-    +> effects
-    +> map_fst (Generate_closure.f +> deadcode')
+    +> (fun (p, sentinal) ->
+         let p, cps_calls = effects p in
+         p, cps_calls, sentinal)
+    +> fun (p, cps_calls, sentinal) ->
+    deadcode' (Generate_closure.f p), cps_calls, sentinal
```

Comment:
```suggestion
    +> map_fst (effects +> map_fst (Generate_closure.f +> deadcode'))
```

Reply:
This change seems to cause a type error with the (very confusing) input type to `emit` from essentially different associations in a product type

Reply:
Sure. You need to adjust function `generate` as well: `(((p, live_vars), cps_calls), sentinal)` instead of `((p, live_vars), cps_calls, sentinal)`.

Reply:
Okay, [9ab2476](https://github.com/ocsigen/js_of_ocaml/pull/1503/commits/9ab2476a25d2fe8c055af90c57ef0b0a753e9270)